sdk/src/nonce_instruction.rs

```diff
@@ -0,0 +1,430 @@
+use crate::{
+    account::{get_signers, KeyedAccount},
+    instruction::{AccountMeta, Instruction, InstructionError},
+    instruction_processor_utils::{limited_deserialize, next_keyed_account, DecodeError},
+    nonce_program::id,
+    nonce_state::{NonceAccount, NonceState},
+    pubkey::Pubkey,
+    system_instruction,
+    sysvar::{
+        recent_blockhashes::{self, RecentBlockhashes},
+        rent::{self, Rent},
+        Sysvar,
+    },
+};
+use log::*;
+use num_derive::{FromPrimitive, ToPrimitive};
+use serde_derive::{Deserialize, Serialize};
+use solana_logger;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, FromPrimitive, ToPrimitive)]
+pub enum NonceError {
```

Comment:
Take `thiserror` for a spin?

Reply:
I'll take a peek at https://github.com/solana-labs/solana/pull/7177